toc/rfc/rfc-draft-hash-based-routing.md

```diff
@@ -0,0 +1,186 @@
+      load situation will not be considered
+    - To account for overload situations, `hash_balance` values should be greater than 110. During the implementation
+      phase, the values will be evaluated to identify the best fit for the recommended range
+    - For load balancing algorithms other than hash, the `hash_balance` and `hash_header` properties MUST not be set
+
+An example for manifest with these properties:
+
+```yaml
+version: 1
+applications:
+  - name: test
+    routes:
+      - route: test.example.com
+        options:
+          loadbalancing: hash
+          hash_header: tenant-id
+          hash_balance: 125
+      - route: anothertest.example.com
+        options:
+          loadbalancing: least-connection
+```
+
+The decision to introduce plain keys was influenced by the following points:
+
+- Simple to use
+- It allows for easy addition of more load-balancing-related properties if new requirements arise in the future
+- It complies with
+  the [RFC](https://github.com/cloudfoundry/community/blob/main/toc/rfc/rfc-0027-generic-per-route-features.md#proposal),
+  which states that the map must use strings as keys and can use numbers, strings, and the literals true and false as
+  values
+
+### No Changes Required
+
+#### CF cli
+
+The [current implementation of route option](https://github.com/cloudfoundry/cli/blob/main/resources/options_resource.go)
+supports the use of `--option KEY=VALUE`, where the key and value are sent directly to CC for validation. Consequently,
+the `create-route`, `update-route`, and `map-route` commands require no modifications, as they already accept the
+proposed properties.
+Example:
+
+```bash
+cf create-route MY-APP example.com -n test -o loadbalancing=hash -o hash_header=tenant-id -o hash_balance=125
+cf update-route MY-APP example.com -n test -o loadbalancing=hash -o hash_header=tenant-id -o hash_balance=125
+cf update-route MY-APP example.com -n test -o loadbalancing=hash -o hash_header=tenant-id
+cf update-route MY-APP example.com -n test -o loadbalancing=hash -o hash_balance=125
+cf map-route MY-APP example.com -n test -o loadbalancing=hash -o hash_header=tenant-id -o hash_balance=125
+```
+
+#### BBS
+
+The route object is maintained as a generic JSON object in the BBS, likely because the BBS doesn't use the route
+information itself. Therefore, it simply accepts and stores the route options provided by the Cloud Controller.
+
+#### Route-Emitter
+
+The options are raw JSON and will be passed directly to the Gorouter without any modifications.
+
+#### Route-Registrar
+
+In the scope of this RFC, it is not planned to implement hash-based routing in route-registrar for platform-routes.
+
+### Diagrams
+
+#### An activity diagram for routing decision for an incoming request
+
+![](rfc-draft-hash-based-routing/ActivityDiagram.drawio.png)
+
+#### A simplified activity diagram for Gorouter's endpoint registration process
+
+![](rfc-draft-hash-based-routing/EndpointRegistration.drawio.png)
```

Comment:
Right column, first box says "Reguster", should say "Register".

Condition decision (yes, no) labels missing in the one 'if' in the diagram.